src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1337,6 +1337,12 @@ EnsureTableNotDistributed(Oid relationId)
 static void
 EnsureRelationHasNoTriggers(Oid relationId)
 {
+	if (EnableUnsafeTriggers)
+	{
+		/* user really wants triggers */
+		return;
+	}
```

Comment:
With this change, I think the name of the function doesn't correctly reflect what it does anymore.

Instead of checking `EnableUnsafeTriggers` here, I would suggest doing the following to be more explicit:
```diff
static void
EnsureRelationCanBeDistributed(Oid relationId, Var *distributionColumn,
							   char distributionMethod, uint32 colocationId,
							   char replicationModel, bool viaDeprecatedAPI)
{
	Oid parentRelationId = InvalidOid;

	EnsureTableNotDistributed(relationId);
	EnsureLocalTableEmptyIfNecessary(relationId, distributionMethod, viaDeprecatedAPI);
+++	if (!EnableUnsafeTriggers)
+++	{
+++		EnsureRelationHasNoTriggers(relationId);
+++	}
---	EnsureRelationHasNoTriggers(relationId);
...
```